controllers/prep_handlers.go

```diff
@@ -69,22 +69,49 @@ func (r *ImageBasedUpgradeReconciler) getSeedImage(
 		return fmt.Errorf("failed to pull image: %w", err)
 	}
 
-	r.Log.Info("Checking seed image compatibility")
-	if err := r.checkSeedImageCompatibility(ctx, ibu.Spec.SeedImageRef.Image); err != nil {
+	labels, err := r.getLabelsForSeedImage(ibu.Spec.SeedImageRef.Image)
+	if err != nil {
+		return fmt.Errorf("failed to get seed image labels: %w", err)
+	}
+
+	r.Log.Info("Checking seed image version compatibility")
+	if err := checkSeedImageVersionCompatibility(labels); err != nil {
+		return fmt.Errorf("checking seed image compatibility: %w", err)
+	}
+
+	seedInfo, err := getSeedConfigFromLabel(labels)
+	if err != nil {
+		return fmt.Errorf("failed to get seed cluster info from label: %w", err)
+	}
+
+	hasProxy, err := lcautils.HasProxy(ctx, r.Client)
+	if err != nil {
+		return fmt.Errorf("failed to check if cluster has proxy: %w", err)
+	}
+
+	r.Log.Info("Checking seed image proxy compatibility")
+	if err := checkSeedImageProxyCompatibility(seedInfo.HasProxy, hasProxy); err != nil {
 		return fmt.Errorf("checking seed image compatibility: %w", err)
 	}
 
 	return nil
 }
 
-// checkSeedImageCompatibility checks if the seed image is compatible with the
-// current version of the lifecycle-agent by inspecting the OCI image's labels
-// and checking if the specified format version equals the hard-coded one that
-// this version of the lifecycle agent expects. That format version is set by
-// the lca-cli during the image build process, and is only manually bumped by
-// developers when the image format changes in a way that is incompatible with
-// previous versions of the lifecycle-agent.
-func (r *ImageBasedUpgradeReconciler) checkSeedImageCompatibility(_ context.Context, seedImageRef string) error {
+func getSeedConfigFromLabel(labels map[string]string) (*seedclusterinfo.SeedClusterInfo, error) {
+	seedFormatLabelValue, ok := labels[common.SeedClusterInfoOCILabel]
```

Comment:
Is this change "beaking" all existing seed images?

Reply:
Yes. I will make it a bit more backwards compatible